Make deprecation notice link to website downloads

albertviilik opened the pull request:
## Description

*This PR updates the ```_printDeprecationWarning()``` method in the ```FlutterCommand``` class to make the deprecation notice output in the terminal by the command ```flutter version``` contain a reference to the https://flutter.dev/docs/development/tools/sdk/releases page where Flutter releases can be downloaded as pre-built archives.*

*The output before the change was:*
```zsh
[!] The "version" command is deprecated and will be removed in a future version of Flutter.

1.20.2
1.21.0-9.0.pre
1.20.1
...
0.0.6
v0.0.6
v0.0.20-alpha
```

*The output after the change will be:*
```zsh
[!] The "version" command is deprecated and will be removed in a future version of Flutter. Refer to https://flutter.dev/docs/development/tools/sdk/releases

1.20.2
1.21.0-9.0.pre
1.20.1
...
0.0.6
v0.0.6
v0.0.20-alpha
```

## Related Issues

Fixes #63787 

## Tests

I added the following tests:

* Updated the ```version ls``` test to check if the output links to the website downloads.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

albertviilik committed:
Test

Tests if the `flutter version` deprecation notice links to the website downloads.

albertviilik committed:
Updated the Deprecation Notice

Added the requested link to the deprecation notice that would link to the website downloads.

jonahwilliams commented on packages/flutter_tools/lib/src/runner/flutter_command.dart:
```diff
@@ -795,7 +795,8 @@ abstract class FlutterCommand extends Command<void> {
       globals.printStatus('$warningMark The "$name" command is deprecated and '
-          'will be removed in a future version of Flutter.');
+          'will be removed in a future version of Flutter. '
+          'Refer to https://flutter.dev/docs/development/tools/sdk/releases');
```
Maybe: "See https://flutter.dev/docs/development/tools/sdk/releases for previous releases of flutter"

albertviilik replied:
Thank you for the feedback! Your version is better. Will use it instead of the initially chosen one.

albertviilik committed:
Changed last sentence

Made the last sentence in the deprecation notice more verbose as recommended by a reviewer and updated the test accordingly.

jonahwilliams commented on packages/flutter_tools/lib/src/runner/flutter_command.dart:
```diff
@@ -796,7 +796,8 @@ abstract class FlutterCommand extends Command<void> {
           'will be removed in a future version of Flutter. '
-          'Refer to https://flutter.dev/docs/development/tools/sdk/releases');
+          'See https://flutter.dev/docs/development/tools/sdk/releases '
+          'for previous releases of flutter.');
```
```suggestion
          'for previous releases of Flutter.');
```

jonahwilliams reviewed (approved):
LGTM, thank you for fixing this!

Commit:
Update packages/flutter_tools/lib/src/runner/flutter_command.dart

Capitalised "Flutter"

Co-authored-by: Jonah Williams <jonahwilliams@google.com>

jonahwilliams commented on packages/flutter_tools/test/commands.shard/hermetic/version_test.dart:
```diff
@@ -43,7 +43,10 @@ void main() {
+        '[!] The "version" command is deprecated '
+            'and will be removed in a future version of Flutter. '
+            'See https://flutter.dev/docs/development/tools/sdk/releases '
+            'for previous releases of flutter.\n\n'
```
```suggestion
            'for previous releases of Flutter.\n\n'
```

Commit:
Update packages/flutter_tools/test/commands.shard/hermetic/version_test.dart

christopherfujino commented:
@albertviilik thanks again for fixing this!
